src/rgw/rgw_op.cc

```diff
@@ -901,7 +901,10 @@ void rgw_build_iam_environment(rgw::sal::Store* store,
     s->env.emplace("aws:UserAgent", i->second);
   }
 
-  i = m.find("QUERY_STRING"); {
+  i = m.find("QUERY_STRING");
+  //["QUERY_STRING"] = "", will be always present in the api request even it is empty, 
+  //but for UT this entry may not be there, so adding if check.
```

Comment:
comment should say :
// Handle non presence of "QUERY_STRING"

Reply:
Done